### DISCUSSION TITLE
Interpreting experiment results

### DISCUSSION THREAD

**Discussion by artizzq** [GeoChemFoam/GeoChemFoam#6] [Q&A] [answered]
I followed your Tutorial 5 on making heat transfer.

I used black and white images of porous media of my own (stack of it) in RAW format zipped in tar.gz

Could you tell me why my resulting temperature field is done towards the center of the cube. It looks like it doesnt respect any local particularities.

![image](https://user-images.githubusercontent.com/56273839/190117927-d9afd57d-50bf-4ffa-ae15-4b15abde5d39.png)

The same happens when I just tried to invert my image

![image](https://user-images.githubusercontent.com/56273839/190118085-64531a76-c6a5-48c3-afc3-476c8bb44719.png)

Is it normal or its not normal?

**Comment by artizzq** [comment]
I tried with data downloaded from the net (for example: Bentheimer_2d25um_binary, https://www.digitalrocksportal.org/projects/317/origin_data/1360/)

and got same result
![image](https://user-images.githubusercontent.com/56273839/190249128-888b6a11-62aa-489e-8715-653299413209.png)


initCase.sh modified according to the data:
```
#!/bin/bash

###### USERS INPUT ############################################################

#Define image name
Image_name="Bentheimer_2d25um_binary"

# Define image dimensions
x_dim=500
y_dim=1000
z_dim=1000

# define resolution (m)
res=0.000025

#padding for inlet/outlet
padWidth=0

#Values of solid, pore, and minimum porosity value for the solid phase (note: this CANNOT be 0)
pores_value=0
solid_value=255
eps_min=0.5

# Define cropping parameters
x_min=0
x_max=200
y_min=0
y_max=250
z_min=0
z_max=250

# number of cells of initial mesh
n_x=50
n_y=50
n_z=50

#Mesh refinement level
nlevel=1
refineStokes=0

#Smoothing parameters: smooth surface when image has artifical roughness created ny segmentation to avoid error when using adaptive mesh
nSmooth=0
cSmooth=4

# flow direction 0 or 1, 2 is empty
direction=0

# Number of processors
NP=8

#### END OF USER INPUT #######################################################

#Insert dimensions in postProcessDict
x_1=0
y_1=0
z_1=0

xSize=$(echo "$x_max - $x_min" | bc)
ySize=$(echo "$y_max - $y_min" | bc)
zSize=$(echo "$z_max - $z_min" | bc)

x_2=$(expr $xSize*$res | bc)
y_2=$(expr $ySize*$res | bc)
z_2=$(expr $zSize*$res | bc)

cp system/postProcessDict1 system/postProcessDict
sed -i "s/x_1/$x_1/g" system/postProcessDict
sed -i "s/y_1/$y_1/g" system/postProcessDict
sed -i "s/z_1/$z_1/g" system/postProcessDict

sed -i "s/x_2/$x_2/g" system/postProcessDict
sed -i "s/y_2/$y_2/g" system/postProcessDict
sed -i "s/z_2/$z_2/g" system/postProcessDict

sed -i "s/flowdir/$direction/g" system/postProcessDict

mkdir constant/polyMesh
cp system/blockMeshDict$direction constant/polyMesh/blockMeshDict

mkdir constant/triSurface
cd constant/triSurface
cp $GCFOAM_RUNS/Custom/$Image_name\.raw.tar.gz .
tar -xf $Image_name.raw.tar.gz
cd ../..

#Dummy fluid properties
flowRate=1e-40 #1e-30

cp -r 0_orig 0

sed -i "s/flow_rate/$flowRate/g" 0/U

cp system/fvSolution1 system/fvSolution
sed -i "s/nSmooth/1/g" system/fvSolution
sed -i "s/cSmooth/0.5/g" system/fvSolution

python createblockmesh.py —xDim $x_dim —yDim $y_dim —zDim $z_dim —xMin $x_min —xMax $x_max —yMin $y_min —yMax $y_max —zMin $z_min —zMax $z_max —nX $n_x —nY $n_y —nZ $n_z —nLevel $nlevel —refineStokes $refineStokes —res $res —Image_name $Image_name —padWidth $padWidth —pores_value $pores_value —solid_value $solid_value —eps_min $eps_min —direction $direction —NP $NP

rm constant/triSurface/$Image_name.raw

cp system/fvSolution1 system/fvSolution
sed -i "s/nSmooth/$nSmooth/g" system/fvSolution
sed -i "s/cSmooth/$cSmooth/g" system/fvSolution

smoothSolidSurface > smoothSolidSurface.out

echo -e "Case initialised. It is advised to check in paraview to confirm mesh and 0/eps are reasonable before running flow"
```

Is it something with algorithm that sticked to some cube resolution?

**Comment by GeoChemFoam** [comment]
It looks to me like your mesh has failed. Change y_max and z_max to 200. (250/50/n_level+1 is not an integer).  Furthermore, you have changed many of the default values in the initcase script to values that don't make any sense. An eps_min of 0.5 means that the porosity of your solid will be set to 50% (i.e. not solid). Change it back to 0.0001. Also, put the smoothing parameters back to nSmooth=1, cSmooth=0.1 and try again. Always load your mesh into paraview before running flow/heat. If it doesn't look like a porous media then the meshing has failed and you will need to change something.
________________________________

Founded in 1821, Heriot-Watt is a leader in ideas and solutions. With campuses and students across the entire globe we span the world, delivering innovation and educational excellence in business, engineering, design and the physical, social and life sciences. This email is generated from the Heriot-Watt University Group, which includes:

  1.  Heriot-Watt University, a Scottish charity registered under number SC000278
  2.  Heriot- Watt Services Limited (Oriam), Scotland's national performance centre for sport. Heriot-Watt Services Limited is a private limited company registered is Scotland with registered number SC271030 and registered office at Research & Enterprise Services Heriot-Watt University, Riccarton, Edinburgh, EH14 4AS.

The contents (including any attachments) are confidential. If you are not the intended recipient of this e-mail, any disclosure, copying, distribution or use of its contents is strictly prohibited, and you should please notify the sender immediately and then delete it (including any attachments) from your system.

**Comment by GeoChemFoam** [comment]
Note that your voxel size is also incorrect. It should be 2.25microns not 25.
________________________________

Founded in 1821, Heriot-Watt is a leader in ideas and solutions. With campuses and students across the entire globe we span the world, delivering innovation and educational excellence in business, engineering, design and the physical, social and life sciences. This email is generated from the Heriot-Watt University Group, which includes:

  1.  Heriot-Watt University, a Scottish charity registered under number SC000278
  2.  Heriot- Watt Services Limited (Oriam), Scotland's national performance centre for sport. Heriot-Watt Services Limited is a private limited company registered is Scotland with registered number SC271030 and registered office at Research & Enterprise Services Heriot-Watt University, Riccarton, Edinburgh, EH14 4AS.

The contents (including any attachments) are confidential. If you are not the intended recipient of this e-mail, any disclosure, copying, distribution or use of its contents is strictly prohibited, and you should please notify the sender immediately and then delete it (including any attachments) from your system.

**Comment by artizzq** [comment]
I followed you instructions and got the same result :(
![image](https://user-images.githubusercontent.com/56273839/190450880-332aad23-221c-462b-a2d9-e7d504a3579f.png)
```
#!/bin/bash

###### USERS INPUT ############################################################

#Define image name
Image_name="Bentheimer_2d25um_binary"

# Define image dimensions
x_dim=1000
y_dim=1000
z_dim=1000

# define resolution (m)
res=0.00000225

#padding for inlet/outlet
padWidth=0

#Values of solid, pore, and minimum porosity value for the solid phase (note: this CANNOT be 0)
pores_value=0
solid_value=255
eps_min=0.0001

# Define cropping parameters
x_min=0
x_max=200
y_min=0
y_max=200
z_min=0
z_max=200

# number of cells of initial mesh
n_x=50
n_y=50
n_z=50

#Mesh refinement level
nlevel=0
refineStokes=0

#Smoothing parameters: smooth surface when image has artifical roughness created ny segmentation to avoid error when using adaptive mesh
nSmooth=1
cSmooth=0.1

# flow direction 0 or 1, 2 is empty
direction=0

# Number of processors
NP=8

#### END OF USER INPUT #######################################################

#Insert dimensions in postProcessDict
x_1=0
y_1=0
z_1=0
```

**Comment by GeoChemFoam** [comment]
Open the mesh in paraview. You are showing T

On Sep 15, 2022, at 16:57, artizzq ***@***.***> wrote:

﻿
****************************************************************
Caution: This email originated from a sender outside Heriot-Watt University.
Do not follow links or open attachments if you doubt the authenticity of the sender or the content.
****************************************************************


I followed you instructions and got the same result :(
[image]<https://user-images.githubusercontent.com/56273839/190450880-332aad23-221c-462b-a2d9-e7d504a3579f.png>

#!/bin/bash

###### USERS INPUT ############################################################

#Define image name
Image_name="Bentheimer_2d25um_binary"

# Define image dimensions
x_dim=1000
y_dim=1000
z_dim=1000

# define resolution (m)
res=0.00000225

#padding for inlet/outlet
padWidth=0

#Values of solid, pore, and minimum porosity value for the solid phase (note: this CANNOT be 0)
pores_value=0
solid_value=255
eps_min=0.0001

# Define cropping parameters
x_min=0
x_max=200
y_min=0
y_max=200
z_min=0
z_max=200

# number of cells of initial mesh
n_x=50
n_y=50
n_z=50

#Mesh refinement level
nlevel=0
refineStokes=0

#Smoothing parameters: smooth surface when image has artifical roughness created ny segmentation to avoid error when using adaptive mesh
nSmooth=1
cSmooth=0.1

# flow direction 0 or 1, 2 is empty
direction=0

# Number of processors
NP=8

#### END OF USER INPUT #######################################################

#Insert dimensions in postProcessDict
x_1=0
y_1=0
z_1=0


—
Reply to this email directly, view it on GitHub<https://github.com/GeoChemFoam/GeoChemFoam/discussions/6#discussioncomment-3655395>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AWDOAFQPIS4KLVG23YTWXL3V6NBQHANCNFSM6AAAAAAQMHGF6U>.
You are receiving this because you commented.Message ID: ***@***.***>

________________________________

Founded in 1821, Heriot-Watt is a leader in ideas and solutions. With campuses and students across the entire globe we span the world, delivering innovation and educational excellence in business, engineering, design and the physical, social and life sciences. This email is generated from the Heriot-Watt University Group, which includes:

  1.  Heriot-Watt University, a Scottish charity registered under number SC000278
  2.  Heriot- Watt Services Limited (Oriam), Scotland's national performance centre for sport. Heriot-Watt Services Limited is a private limited company registered is Scotland with registered number SC271030 and registered office at Research & Enterprise Services Heriot-Watt University, Riccarton, Edinburgh, EH14 4AS.

The contents (including any attachments) are confidential. If you are not the intended recipient of this e-mail, any disclosure, copying, distribution or use of its contents is strictly prohibited, and you should please notify the sender immediately and then delete it (including any attachments) from your system.

**Comment by GeoChemFoam** [comment, marked as answer]
I found the problem. You have no identified the phases correctly. Open the .raw file in fiji and look at your image. You will find your answer
________________________________

Founded in 1821, Heriot-Watt is a leader in ideas and solutions. With campuses and students across the entire globe we span the world, delivering innovation and educational excellence in business, engineering, design and the physical, social and life sciences. This email is generated from the Heriot-Watt University Group, which includes:

  1.  Heriot-Watt University, a Scottish charity registered under number SC000278
  2.  Heriot- Watt Services Limited (Oriam), Scotland's national performance centre for sport. Heriot-Watt Services Limited is a private limited company registered is Scotland with registered number SC271030 and registered office at Research & Enterprise Services Heriot-Watt University, Riccarton, Edinburgh, EH14 4AS.

The contents (including any attachments) are confidential. If you are not the intended recipient of this e-mail, any disclosure, copying, distribution or use of its contents is strictly prohibited, and you should please notify the sender immediately and then delete it (including any attachments) from your system.

**Comment by artizzq** [comment]
What's wrong with it?
![image](https://user-images.githubusercontent.com/56273839/190456604-c63697ab-ba0b-4493-a000-169297307baf.png)
Comparing to tutorial 5 image, the difference between bitness of images is clearly visible
![image](https://user-images.githubusercontent.com/56273839/190457775-beb2e485-da9f-411a-a922-fdb6d930dc3d.png)
Maybe thats the point?

**Comment by GeoChemFoam** [comment]
What value are the pores? What value are the grains? Now look at what values you have put in the initcase script. Do they match? (hint: no)
<img width="1037" alt="Screenshot 2022-09-15 at 17 39 22" src="https://user-images.githubusercontent.com/42167513/190460040-3bef0936-403d-4175-ac8b-99bb99bf4371.png">
<img width="814" alt="Screenshot 2022-09-15 at 17 39 33" src="https://user-images.githubusercontent.com/42167513/190460067-1cf03bf8-27c5-4445-b525-8ca669ec1720.png">


________________________________

Founded in 1821, Heriot-Watt is a leader in ideas and solutions. With campuses and students across the entire globe we span the world, delivering innovation and educational excellence in business, engineering, design and the physical, social and life sciences. This email is generated from the Heriot-Watt University Group, which includes:

  1.  Heriot-Watt University, a Scottish charity registered under number SC000278
  2.  Heriot- Watt Services Limited (Oriam), Scotland's national performance centre for sport. Heriot-Watt Services Limited is a private limited company registered is Scotland with registered number SC271030 and registered office at Research & Enterprise Services Heriot-Watt University, Riccarton, Edinburgh, EH14 4AS.

The contents (including any attachments) are confidential. If you are not the intended recipient of this e-mail, any disclosure, copying, distribution or use of its contents is strictly prohibited, and you should please notify the sender immediately and then delete it (including any attachments) from your system.

**Comment by artizzq** [comment]
Seems to work now. Yes, values didnt match. Had to check.

![image](https://user-images.githubusercontent.com/56273839/190552701-08fd8d32-b491-4f51-a0fb-d07c72082127.png)

Also one question:
**Is it possible to define start point and dont use entire plane for simulation start?**

**Comment by GeoChemFoam** [comment]
I dont understand this question: "Also one question:
Is it possible to define start point and dont use entire plane for simulation start?”

Do you mean the entire domain? You are already cropping it. This simulation is for steady state heat transfer so there is no start per se.
________________________________

Founded in 1821, Heriot-Watt is a leader in ideas and solutions. With campuses and students across the entire globe we span the world, delivering innovation and educational excellence in business, engineering, design and the physical, social and life sciences. This email is generated from the Heriot-Watt University Group, which includes:

  1.  Heriot-Watt University, a Scottish charity registered under number SC000278
  2.  Heriot- Watt Services Limited (Oriam), Scotland's national performance centre for sport. Heriot-Watt Services Limited is a private limited company registered is Scotland with registered number SC271030 and registered office at Research & Enterprise Services Heriot-Watt University, Riccarton, Edinburgh, EH14 4AS.

The contents (including any attachments) are confidential. If you are not the intended recipient of this e-mail, any disclosure, copying, distribution or use of its contents is strictly prohibited, and you should please notify the sender immediately and then delete it (including any attachments) from your system.
